File: src/test/java/com/joecollins/graphics/utils/BindableWrapper.java
package com.joecollins.graphics.utils;

import com.joecollins.bindings.Bindable;
import com.joecollins.bindings.Binding;

public class BindableWrapper<T> extends Bindable {
  private enum BindableWrapperValue {
    VALUE
  }

  private T value;

  public BindableWrapper() {}

  public BindableWrapper(T value) {
    setValue(value);
  }

  public T getValue() {
    return value;
  }

  public void setValue(T value) {
    this.value = value;
    onPropertyRefreshed(BindableWrapperValue.VALUE);
  }

  public Binding<T> getBinding() {
    return Binding.propertyBinding(this, BindableWrapper::getValue, BindableWrapperValue.VALUE);
  }
}

File: src/test/java/com/joecollins/graphics/utils/BindableWrapper.kt
package com.joecollins.graphics.utils

import com.joecollins.bindings.Bindable
import com.joecollins.bindings.Binding
import com.joecollins.graphics.utils.BindableWrapper.BindableWrapperValue

class BindableWrapper<T>(private var _value: T) : Bindable<BindableWrapper<T>, BindableWrapperValue>() {

    enum class BindableWrapperValue {
        VALUE
    }

    var value: T
    get() = _value
    set(value) {
        this._value = value
        onPropertyRefreshed(BindableWrapperValue.VALUE)
    }

    val binding: Binding<T>
        get() = Binding.propertyBinding(this, { it.value }, BindableWrapperValue.VALUE)
}
